docs/snippets/angular/component-story-custom-args-icons.ts.mdx

```diff
@@ -0,0 +1,37 @@
+```ts
+// YourComponent.stories.ts
+
+import { Story, Meta } from '@storybook/angular/types-6-0';
+import Icon from './icon.component';
+
+import { IconA, IconB, IconC, IconD, IconE } from './icons';
+
+// Maps the icons to a JSON serializable object to be safely used with the argTypes
+const iconMap = { IconA, IconB, IconC, IconD, IconE };
+
+export default {
+  title: 'My Story with Icons',
+  component: YourComponent,
+  argTypes: {
+    icon: {
+      control: {
+        type: 'select',
+        options: Object.keys(iconMap),
+      },
+    },
+  },
+} as Meta;
+
+const Template: Story<Icon> = (args: Icon) => {
+  // retrieves the appropriate icon passes it as a component prop
+  const { icon } = args;
+  const selectedIcon = iconMap[icon];
+  return {
+    component: Icon,
```

Comment:
should we not change : `component: YourComponent,` to `component: Icon,`
and remove this line ? 

if I am not mistaken it is preferable that users create one story per component. no ?